### ISSUE TITLE
Codemirror typing issue

### ISSUE THREAD

**Issue by martinRenou** [jupyterlab/jupyterlab#9295]
## Description

I faced this issue when trying to update a codemirror extension for JupyterLab 3: https://github.com/robocorp/robocode-kernel/pull/6

The line where the compilation issue appears is https://github.com/robocorp/robocode-kernel/pull/6/files#diff-53be0f8e096ce0a39e0d76d884da27af721ce724a1bea5eba8a1029b78ee330bR586

The problem is when calling `CodeMirror.defineSimpleMode`, passing a `ISimpleMeta<string, any>` object, TypeScript complains with:

```
src/mode.ts:586:5 - error TS2559: Type 'ISimpleMeta<string, any>' has no properties in common with type 'ISimpleMode<string>'.

586     meta
        ~~~~

  node_modules/@jupyterlab/codemirror/typings/codemirror/codemirror.d.ts:117:5
    117     meta: ISimpleMode<S>;
            ~~~~
    The expected type comes from property 'meta' which is declared here on type 'Partial<TSimpleStates<string, any> & ISimpleMeta<string, any>>'


Found 1 error.
```

## Reproduce

Call `CodeMirror.defineSimpleMode` passing a `ISimpleMeta<string, any>` object.

## Context

- @jupyterlab/codemirror: "^3.0.0-rc.7"
- JupyterLab version: 3

<details><summary>Troubleshoot Output</summary>
<pre>
Paste the output from running `jupyter troubleshoot` from the command line here.
You may want to sanitize the paths in the output.
</pre>
</details>

<details><summary>Command Line Output</summary>
<pre>
Paste the output from your command line running `jupyter lab` here, use `--debug` if possible.
</pre>
</details>

<details><summary>Browser Output</summary>
<pre>
Paste the output from your browser Javascript console here.
</pre>
</details>

**Comment by jtpio** [comment]
We looked into it a bit with Martin on CodeSandbox to have a smaller reproducible example:

https://codesandbox.io/s/jupyterlab-codemirror-typings-forked-27fdc?file=/src/index.ts

**Comment by jtpio** [comment]
cc @bollwyvl who might have some pointers, as the types were added in https://github.com/jupyterlab/jupyterlab/pull/9123

**Comment by bollwyvl** [comment]
Yeah, sorry. Turns out those typings you're adding (which i wrote, ha!) are already up in lab core.

This [PR to extension-examples](https://github.com/jupyterlab/extension-examples/pull/130) shows how the typings can/could be used more-or-less directly... but a number of dependency resolution things changed in the intervening torrent of rc releases, so I'm not sure if it does what it says on the tin any longer... will kick the tires on that PR vs rc13 (or whatever we're up to now).

**Comment by krassowski** [comment]
I think this can be closed as JupyterLab 3 (no longer supported) was using CodeMirror 5 and JupyterLab 4+ uses CodeMirror 6.
